compel/plugins/std/log.c

```diff
@@ -51,11 +54,19 @@ static void sbuf_log_init(struct simple_buf *b)
 	 */
 	b->bp = b->buf;
 
-	if (start.tv_sec != 0) {
+	if (start.tv_sec != 0 || print_ts_diffs) {
 		struct timeval now;
 
 		sys_gettimeofday(&now, NULL);
-		timediff(&start, &now);
+		struct timeval *pivot = print_ts_diffs ? &last_t : &start;
```

Comment:
all variables have to be declared at the beginning of a block where they are used.